common/src/main/scala/org/neo4j/spark/service/SchemaService.scala

```diff
@@ -250,6 +250,80 @@ class SchemaService(private val options: Neo4jOptions, private val driverCache:
     StructType(sortedStructFields)
   }
 
+  def structForGDS(): StructType = {
+    val query =
+      """
+        |CALL gds.list() YIELD name, signature, type
+        |WHERE name = $procName AND type = 'procedure'
+        |WITH split(signature, ') :: (')[1] AS fields
+        |WITH substring(fields, 0, size(fields) - 1) AS fields
+        |WITH split(fields, ',') AS fields
+        |WITH [field IN fields | split(field, ' :: ')] AS fields
+        |UNWIND fields AS field
+        |WITH field
+        |RETURN *
+        |""".stripMargin
+    val map: util.Map[String, AnyRef] = Map[String, AnyRef]("procName" -> options.query.value).asJava
```

Comment:
nit: it could be nice to expose `procName` as a parameter maybe, so that it's symmetric with `inputForGDSProc`?

Reply:
It's already a parameter. Maybe I'm missing something?

Reply:
The `structForGDS` method does take any parameters
The `inputForGDSProc` method takes the proc name as parameter
I was wondering if we could bring them more in line, but I guess not

Reply:
Oh, I see it now, they have different purposes:
* `structForGDS` retrieves the fields `YIELD` by the procedure that we need to compute the structure of the Dataframe
* `inputForGDSProc` has a different purpose, we need it in order to compute the input parameters required by the procedure so we can map the .option("gds.*", <value>) to each one and eventually complain about some missing field if the user does not provide it